spec/models/claim_update_spec.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ClaimUpdate do
+  let(:claim_update) { ClaimUpdate.new(claim, params, context) }
+
+  context "with parameters that are valid for the context" do
+    let(:claim) { create(:tslr_claim) }
+    let(:context) { "full-name" }
+    let(:params) { {full_name: "Ada Lovelace"} }
+
+    it "updates the claim and returns a truthy value" do
+      expect(claim_update.perform).to be_truthy
+      expect(claim.reload.full_name).to eq "Ada Lovelace"
+    end
+  end
+
+  context "with parameters that are not valid for the context" do
+    let(:claim) { create(:tslr_claim) }
+    let(:context) { "full-name" }
+    let(:params) { {full_name: nil} }
```

Comment:
I would expect this to be testing unrelated parameter changes, e.g. `{postcode: "TE5 T42"}`. Perhaps it could be better named, and we could reject changes to unrelated fields? See https://trello.com/c/EvX3BYxI/219-only-allow-the-expected-values-for-a-given-page

Reply:
Ah, perhaps the spec isn't clear then. It's meant to be verifying that validation that is specific to the the context gets fired, i.e. if they're updating on the "full-name" page, we'd expect the validation that requires the full-name be present in that context fires. Is there a way I could make that clearer? To be fair I probably shouldn't be using the full-name context anyway as it will disappear shortly when Verify is enabled...

Regarding that other Trello card, I think we should tackle that as a separate piece of work if we decide to do it, as it'll require some more major changes to define and honour the parameters that are editable in each context.

Reply:
> Ah, perhaps the spec isn't clear then...

I think it's just that word "valid" again. Maybe this would be clearer rephrased to something like: "without required parameters for the context"?

> Regarding that other Trello card...

Yes agreed. I just thought I'd bring it up as some context on what my line of thinking was.

Reply:
OK, I reworded the spec as suggested. I've also switched to use a context that isn't about to be removed so we won't have to update this spec later.